### DISCUSSION TITLE
Any documentation for serialization?

### DISCUSSION THREAD

**Discussion by weifenluo** [VerifyTests/Verify#852] [Q&A] [answered]
Is there any documentation for how the object to be verified get serialized to the file? Is there any API to customize the serialization behavior? I'm a new comer to this project, for me it's like a magic black box.

**Comment by SimonCropp** [comment, marked as answer]
hows this https://github.com/VerifyTests/Verify/blob/main/docs/serializer-settings.md#serializer-settings

**Comment by KenBonny** [comment]
I'm using F# to write my tests and I want to have strict JSON as output, but F# doesn't support the `ModuleInitializer` attribute. I haven't found how I can do this in F# yet. When I put this line at the start of my test, it generates errors: The API 'TestName@33.MoveNext' must be called prior to any Verify has run. Usually this is done in a [ModuleInitializer].

Is there another way of specifying strict JSON?

**Reply by SimonCropp** [reply]
what test framework are you using?

**Reply by KenBonny** [reply]
I'm using xUnit as my test framework, latest version.

**Reply by SimonCropp** [reply]
try this 

```
public abstract class TestsBase
{
    static TestsBase()
    {
        // Do "global" initialization here
    }
}

public class MyTests : TestsBase
{
    // Add test methods
}
```

**Reply by KenBonny** [reply]
I was so caught up in writing functional code, I forgot about this. The base class does not alleviate the problem as the `VerifierSettings` get called each test run, which triggers the error. Using a `ClassFixture` fixed it though. The class fixture is instantiated once and thus the setup is ran once. Works like a charm. Thank you for pointing me in the right direction. 💖

```fsharp
type VerifierSettingsFixture() =
    do VerifierSettings.UseStrictJson()
    interface IDisposable with
        member this.Dispose() = ()

[<UsesVerify>]
type ``The test class``(verifierSettingsFixture: VerifierSettingsFixture) =
    interface IClassFixture<VerifierSettingsFixture>
    // tests go here
```
